pkg/server/datastore/sqlstore/sqlstore.go

```diff
@@ -1061,9 +1061,15 @@ func (ds *Plugin) openDB(cfg *configuration, isReadOnly bool) (*gorm.DB, string,
 	if cfg.MaxOpenConns != nil {
 		db.DB().SetMaxOpenConns(*cfg.MaxOpenConns)
 	}
+	db.DB().SetMaxIdleConns(100) // default value
 	if cfg.MaxIdleConns != nil {
 		db.DB().SetMaxIdleConns(*cfg.MaxIdleConns)
 	}
+	defaultConnMaxLifetime, err := time.ParseDuration("30s")
+	db.DB().SetConnMaxLifetime(defaultConnMaxLifetime) // default value
+	if err != nil {
+		return nil, "", false, nil, fmt.Errorf("failed to parse conn_max_lifetime %q: %w", defaultConnMaxLifetime, err)
+	}
```

Comment:
The SetConnMaxLifetime function is already called later, based on the ConnMaxLifetime setting.
We don't need to make any change there. What we need to adjust is the ConnMaxIdleTime property calling SetConnMaxIdleTime, maybe with a value of 30s.
Also note that you will not need to make any parsing for that (i.e. `time.ParseDuration("30s")`), you can just call `SetConnMaxIdleTime(time.Second * 30`). We may define this as a constant in the function, and probably the MaxOpenConns and MaxIdleConns numbers too.